src/runtime/vm/executable.cc

```diff
@@ -476,8 +484,19 @@ void LoadHeader(dmlc::Stream* strm) {
 }
 
 runtime::Module Executable::Load(const std::string& code, const runtime::Module lib) {
+  std::cout << "code: " << code.size() << std::endl;
   auto exec = make_object<Executable>();
-  exec->lib = lib;
+
+  // Support null-initialization of lib, to enable initialization during
+  // deserialization before we have we have deserialized the imports.
+  if (lib.defined()) {
```

Comment:
perhaps it's worth either creating a const bool is_initialized() { ICHECK_LE(this->imports_.size(), 1); return this->imports_.size() == 1; } and adding it everywhere we access lib, or make the change I suggested above in GetLib and audit all uses of lib to ensure we GetLib always.

Reply:
The problem is in this case we actually need to check the argument, arguably it should be like Optional<runtime::Module> but since Module is nullable you have to check for inner nullability either way. I don't think that would clean up this use site, I made the change in GetLib and will make that and SetLib preferred interface at least until we figure out aritfact situation.